### DISCUSSION TITLE
eslint commit-analyzer preset potentially not working, missing configuration, or missing logging

### DISCUSSION THREAD

**Discussion by alisters** [semantic-release/semantic-release#2941] [Configuration] [answered]
### Current behavior

I'm testing out semantic-release, and would like to use the eslint commit format. I'm trying to get a very simple config going. Unfortunately i don't have a lot of experience with node and npm, so i might be overlooking some obvious. If i use the following config, i see the commits being analyzed, but as expected none matched - because my commits messages aren't in that format !

in .releaserc.json
``
{
  "branches": ["main"],
  "plugins": [
    [
      "@semantic-release/commit-analyzer",
      {
        "preset": "angular"
      }
    ]
  ]
}
``
Output: 
...
[9:21:35 pm] [semantic-release] » ℹ  Start step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[9:21:35 pm] [semantic-release] [@semantic-release/commit-analyzer] » ℹ  Analyzing commit: New: Try again with default rules (refs #39039)
  semantic-release:commit-analyzer Analyzing with default rules +0ms
[9:21:35 pm] [semantic-release] [@semantic-release/commit-analyzer] » ℹ  The commit should not trigger a release
...

If i change the angular to "eslint" i only see, which should match my format
...
[9:20:17 pm] [semantic-release] » ℹ  Start step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"

No further logging is produced and the command exits. 

My package.json is 
``
{
  "name": "semantic-release-gb",
  "scripts": {
    "semantic-release": "semantic-release"
  },
  "dependencies": {
    "conventional-changelog-eslint": "^5.0.0",
    "semantic-release": "^21.1.1"
  }
}
``
so, i believe i've installed the package for the preset, covering off the comments here about installing either plugins, or presets
https://github.com/semantic-release/semantic-release/blob/master/docs/usage/plugins.md#additional-plugins and https://github.com/semantic-release/commit-analyzer

> Notes: in order to use a preset it must be installed (for example to use the [eslint preset](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-eslint) you must install it with npm install conventional-changelog-eslint -D)

I've had a fair look through the code, e.g https://github.com/semantic-release/commit-analyzer/blob/master/index.js#L46, where i can see in the angular commit case i'm hitting that line, but in the eslint preset case it appears i'm not

In each case i see the same logging, that indicates the same commits have been found, and i assume, passed to the commit-analyzer (this snippet with --debug flag passed)
[9:20:17 pm] [semantic-release] » ℹ  Found 21 commits since last release
  semantic-release:get-commits Parsed commits: [ { commit: ...

Any idea what i might be doing wrong ? 
Can i confirm that no other config is required ? e.g looking here, https://github.com/semantic-release/commit-analyzer/blob/master/README.md#configuration at the config key  "npm package name of a custom [conventional-changelog](https://github.com/conventional-changelog/conventional-changelog) preset" i am assuming that;s only required IF i'm using a preset that's "custom", i.e outside the list of "angular, atom, codemirror ..... eslint...  " i.e eslint being in that list means the config key is not required ? The notes also so "Note: config will be overwritten by the values of preset. You should use either preset or config, but not both.", so i'm fairly sure having ONLY the "preset: key should be ok. 
What behaviour would manifest if the conventional-changelog-eslint was not being picked up ? (Silent failure or would i see an exception ?)
I've tried the config with and without releaseRules as well. I've tried lifting the configuration from here (https://svdoscience.com/2020-10-31/versioning-with-semantic-release) to no avail

I'm running locally to ease testing with the command
 npm run semantic-release -- --dry-run --ci=false --no-ci
(i've also tried running with npx)

PS: Sincerely hope i haven't wasted peoples time with this and it might prove helpful to other node newbies struggling to configure what looks to be a fantastic addition to pipeline automation 

### Expected behavior

IFF this is not a bug AND i'm not missing something obvious node related
- Improved doco around missing config
- Some more error logging 

### `semantic-release` version

 "semantic-release": "^21.1.1"

### CI environment

gitlab

### Plugins used

Inbuilt only - plus eslint "plugin" - conventional-changelog-eslint": "^5.0.0
{
  "branches": ["main"],
  "plugins": [
    [
      "@semantic-release/commit-analyzer",
      {
        "preset": "eslint"
      }
    ]
  ]
}

### `semantic-release` configuration

{
  "branches": ["main"],
  "plugins": [
    [
      "@semantic-release/commit-analyzer",
      {
        "preset": "eslint"
      }
    ]
  ]
}

### CI logs

[10:12:17 pm] [semantic-release] » ℹ  Running semantic-release version 21.1.1
[10:12:17 pm] [semantic-release] » ✔  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
[10:12:21 pm] [semantic-release] » ‼  Run automated release from branch main on repository https://gitlabserver/remote.git in dry-run mode
[10:12:22 pm] [semantic-release] » ✔  Allowed to push to the Git repository
[10:12:22 pm] [semantic-release] » ℹ  No git tag version found on branch main
[10:12:22 pm] [semantic-release] » ℹ  No previous release found, retrieving all commits
[10:12:22 pm] [semantic-release] » ℹ  Found 21 commits since last release
[10:12:22 pm] [semantic-release] » ℹ  Start step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"

**Comment by alisters** [comment]
My only vague suspicion is that semantic release is not picking the "conventional-changelog-eslint" package due to some misconfiguration on my part (it's in node modules though) Removing it from the package.json, in either dependencies or devDependencies has no effect. Also the comments around installing the presets, as an obvious note, seems to me to indicate that might be a gotcha - and it may fail silently if not found ?

**Comment by travi** [comment, marked as answer]
since the version of the eslint preset that you are installing is the [recently released version with breaking changes](https://github.com/conventional-changelog/conventional-changelog/releases/tag/conventional-changelog-eslint-v5.0.0) that have not yet been handled in semantic-release, it looks like you are hitting a combination of the ongoing #2929 and #2932. it is a good idea to take a look at existing issues when you run into problems, especially the ones that maintainers have pinned.

please either pin your preset to the previous major version or try out the beta version that is making progress toward fixing this issue.

**Reply by alisters** [reply]
Thanks @travi thats fixed it. I saw one of those issues, but at the time didn't have enough knowledge to know it was the same cause, and i spent my time searching for eslint related errors (as angular was working but eslint not). Never mind, sometimes attempting to debug a new tool is a great way to learn it !
